significant_strikes.py

```diff
@@ -0,0 +1,500 @@
+        # Initialize variables to store totals
+        sig_str_landed_op = 0
+        sig_str_attempted_op = 0
+        total_fight_time = 0
+
+        # Iterate through each fight in the DataFrame
+        for index, fight in stats_df.iterrows():
+            if fight['fighter_a_id'] == fighter_id:
+                # Fighter is Fighter A
+                sig_str_landed_op += fight['fighter_b_sig_str_landed']
+                sig_str_attempted_op += fight['fighter_b_sig_str_attempted']
+            elif fight['fighter_b_id'] == fighter_id:
+                # Fighter is Fighter B
+                sig_str_landed_op += fight['fighter_a_sig_str_landed']
+                sig_str_attempted_op += fight['fighter_a_sig_str_attempted']
+            
+            # Sum the total fight time
+            total_fight_time += fight['total_fight_time']
+        
+        # If no fights, return default value
+        if total_fight_time == 0:
+            return self.default_val
+
+        return (sig_str_attempted_op - sig_str_landed_op) / total_fight_time
+    
+
+    def convert_time_to_seconds(self, time_str):
+        """
+        Converts a time string in MM:SS format to total seconds.
+
+        Args:
+            time_str (str): The time string to convert.
+
+        Returns:
+            int: The time in total seconds.
+        """
+
+        if ':' in time_str:
+            minutes, seconds = time_str.split(':')
+            return int(minutes) * 60 + int(seconds)
+        else:
+            return int(time_str)
+        
+
+    def get_round_time(self, outcome_round, outcome_time, round):
+        """
+        Calculates the total fight time in a specific round.
+
+        Args:
+            outcome_round (int): The round number at the end of the fight.
+            outcome_time (str): The time at which the fight ended in the outcome round.
+            round (str): The round for which the time is to be calculated.
+
+        Returns:
+            float: The total fight time in the specified round in minutes.
+        """
+
+        # Convert outcome_time to seconds
+        outcome_time = self.convert_time_to_seconds(outcome_time)
+        
+        # Handle the "overall" case
+        if round == "overall":
+            # Calculate total fight time up to the outcome round
+            total_time = 0
+
+            for r in range(1, outcome_round):
+                total_time += 300  # Add full time for each completed round
+
+            total_time += outcome_time  # Add the time of the last (outcome) round
+
+            return total_time / 60  # Convert to minutes
+        
+        round_num = int(round[1])  # Extract the round number from the string
+
+        # Handle specific rounds
+        if outcome_round == round_num:
+            return outcome_time/60  # Convert to minutes
+        
+        elif outcome_round > round_num:
+            return 300/60  # Full round time
+        
+        else:
+            return 0  # Round did not occur
```

Comment:
I think we should start writing some tests to assert that the dataset is actually correct. Test sample shouldn't be the entire dataset but what I think we could do is take one fighter, index the dataset into just that one fighter's fights, and see how the features evolve for that specific fighter over there fights and check validity from there. You don't have to write the test now but it will get more difficult as we make a bunch of features and if it doesn't work it can be hard to see where it broke at.

Reply:
I agree. i was gonna write tests for my features once you wrote your tests to see how you did it lol.

Reply:
Word I was have trouble because of the fighter_a not being the same fighter_a issue, because I want to apply the df.rolling() function [(here)](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.rolling.html) so that I can get the rolling data and verify it, but not being the same fighter columns every fight makes it difficult. I think I'm going to swap every fighter and their respective columns so that the same fighter is in the fighter_a column then do the rolling from there.